src/inputsource.cpp

```diff
@@ -196,72 +193,127 @@ void InputSource::cleanup()
+            }
+        }
     }
 
-    for(auto annotation : metaData.annotations) {
-        Annotation a;
-        auto core = annotation.access<core::AnnotationT>();
+    if(root.contains("annotations") && root["annotations"].isArray()) {
+
+        size_t offset = 0;
 
-        const size_t offset = global_core.offset;
-        const size_t sample_start = core.sample_start;
+        if(global.contains("core:offset")) {
+            offset = global["offset"].toDouble();
+        }
 
-        if (sample_start < offset)
-            continue;
+        auto annotations = root["annotations"].toArray();
 
-        const size_t rel_sample_start = sample_start - offset;
+        for(auto annotation_ref : annotations) {
+            if(annotation_ref.isObject()) {
+                auto sigmf_annotation = annotation_ref.toObject();
 
-        a.sampleRange = range_t<size_t>{rel_sample_start, rel_sample_start + core.sample_count - 1};
-        a.frequencyRange = range_t<double>{core.freq_lower_edge, core.freq_upper_edge};
-        a.description = QString::fromStdString(core.description);
+                Annotation a;
 
-        annotationList.append(a);
+                const size_t sample_start = sigmf_annotation["core:sample_start"].toDouble();
+
+                if (sample_start < offset)
+                    continue;
+
+                const size_t rel_sample_start = sample_start - offset;
+
+                const size_t sample_count = sigmf_annotation["core:sample_count"].toDouble();
+                a.sampleRange = range_t<size_t>{rel_sample_start, rel_sample_start + sample_count - 1};
+
+                const double freq_lower_edge = sigmf_annotation["core:freq_lower_edge"].toDouble();
+                const double freq_upper_edge = sigmf_annotation["core:freq_upper_edge"].toDouble();
+                a.frequencyRange = range_t<double>{freq_lower_edge, freq_upper_edge};
+
+                if(sigmf_annotation.contains("core:description")) {
+                    static bool warning_shown;
+                    if(!warning_shown) {
+                        QMessageBox msgBox(QMessageBox::Warning, "SigMF Warning", "Support for core:description fields in SigMF annotations is deprectated and will be removed in the future. Please use core:label fields instead.");
+                        msgBox.exec();
+                        warning_shown = true;
+                    }
+
+                    a.description = sigmf_annotation["core:description"].toString();
+                }
+                else {
+                    if(sigmf_annotation.contains("core:label")) {
+                        a.description = sigmf_annotation["core:label"].toString();
+                    }
+                    else if(sigmf_annotation.contains("core:comment")) {
+                        a.description = sigmf_annotation["core:comment"].toString();
+                    }
+                }
```

Comment:
IMO the logic order here should be:

```c++
if(sigmf_annotation.contains("core:label")) {
    a.description = sigmf_annotation["core:label"].toString();
} else if(sigmf_annotation.contains("core:description")) {
    a.description = sigmf_annotation["core:description"].toString();
    print_warning();
} else if(sigmf_annotation.contains("core:comment")) {
    a.description = sigmf_annotation["core:comment"].toString();
}
```
And i'm not really sure about the comment piece, that could result in chaos as comments can be of any length. Maybe use the first _n_ charachters only?